src/main/java/core/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,41 @@
+package core.dao;
+
+import core.db.Storage;
+import core.lib.Dao;
+import core.model.Car;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        Storage.addCar(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        return Storage.cars.stream()
+                .filter(x -> Objects.equals(x.getId(), id))
+                .findFirst();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return Storage.cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        Car old = get(car.getId()).get();
+        Storage.cars.set(Storage.cars.indexOf(old), car);
+        return old;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return Storage.cars.removeIf(x -> x.getId().equals(id));
```

Comment:
```suggestion
        return Storage.cars.removeIf(c -> c.getId().equals(id));
```
in streams usually the first letter of an element is being used (Car - c, Human - h, Book  - b)